src/anyRepoHandlePullRequest.ts

```diff
@@ -36,3 +41,30 @@ export const handlePullRequestPayload = async (payload: WebhookPayloadPullReques
     body: ran.length ? `PR success, ran: ${ran.join(", ")}`: "Success, NOOP",
   }
 }
+
+export type UnPromise<T> = T extends Promise<infer U> ? U : T
+// The return type of generatePRInfo
+export type PRInfo = UnPromise<ReturnType<typeof generatePRInfo>>
+
+const generatePRInfo = async (api: Octokit, payload: WebhookPayloadPullRequest, logger: Logger) => {
```

Comment:
not sure what other handlers use from `payload`, but it might be possible to shift what they need into PRInfo and stop passing in payload at all. It doesn't save any API calls, but it would make the code more cohesive.

Reply:
Yeah, not a bad call, next time I refactor in here - I'll look